conformance/tests/httproute-cross-namespace.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```

Comment:
Can you keep a single test case and add the remaining ones in a separate PR? That would minimize the likelihood of shipping buggy tests. This PR is big as is.

Reply:
Good call, I left the simplest test in and have pulled the others out for a follow up PR.